homeassistant/components/benq_projector/switch.py

```diff
@@ -0,0 +1,185 @@
+"""Use serial protocol of BenQ projector to obtain state of the projector."""
+# Forked on top of the Acer projector integration! https://github.com/home-assistant/core/tree/dev/homeassistant/components/acer_projector
+# Credits = TrianguloY, CloudBurst, Hikaru, YayMuffins
+import logging
+import re
+
+import serial
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA, SwitchEntity
+from homeassistant.const import (
+    CONF_FILENAME,
+    CONF_NAME,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNKNOWN,
+)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BAUDRATE = "baudrate"
+CONF_TIMEOUT = "timeout"
+CONF_WRITE_TIMEOUT = "write_timeout"
+
+DEFAULT_NAME = "BenQ Projector"
+DEFAULT_TIMEOUT = 4
+DEFAULT_WRITE_TIMEOUT = 4
+
+LAMP_MODE = "Lamp Mode"
+
+ICON = "mdi:projector"
+
+INPUT_SOURCE = "Input Source"
+
+LAMP = "Lamp"
+LAMP_HOURS = "Lamp Hours"
+
+MODEL = "Model"
+
+# Commands known to the projector
+#    STATE_OFF: "* 0 IR 002\r",
+CMD_DICT = {
+    LAMP: "\r*pow=?#\r",
+    LAMP_HOURS: "\r*ltim=?#\r",
+    INPUT_SOURCE: "\r*sour=?#\r",
+    LAMP_MODE: "\r*lampm=?#\r",
+    MODEL: "\r*modelname=?#\r",
+    STATE_ON: "\r*pow=on#\r",
+    STATE_OFF: "\r*pow=off#\r",
+}
+
+DEFAULT_BAUDRATE = 115200
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_FILENAME): cv.isdevice,
+        vol.Optional(CONF_BAUDRATE, default=DEFAULT_BAUDRATE): cv.positive_int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(
+            CONF_WRITE_TIMEOUT, default=DEFAULT_WRITE_TIMEOUT
+        ): cv.positive_int,
+    }
+)
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configuration or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1: